packages/swing-store/src/snapStore.js

```diff
@@ -91,21 +95,26 @@ export const fsStreamReady = stream =>
 export function makeSnapStore(
   root,
   {
-    tmpName,
     createReadStream,
     createWriteStream,
     measureSeconds,
+    fsync,
     open,
-    resolve,
     rename,
+    resolve: pathResolve,
     stat,
+    tmpFile,
+    tmpName,
     unlink,
   },
   { keepSnapshots = false } = {},
 ) {
   /** @type {(opts: unknown) => Promise<string>} */
   const ptmpName = promisify(tmpName);
 
+  /** @type {(fd: number) => Promise<void>} */
+  const pfsync = promisify(fsync);
+
   /**
    * Returns the result of calling a function with the name
    * of a temp file that exists only for the duration of
```

Comment:
Would `withTempName` be simpler on top of `aggregateTryFinally`?
```js
  async function withTempName(fn, prefix = 'tmp') {
    const name = await ptmpName({
      tmpdir: root,
      template: `${prefix}-XXXXXX.xss`,
    });
    return aggregateTryFinally(
      () => fn(name),
      // Ignore file deletion errors.
      () => unlink(name).catch(sink),
    );
  }
```
Or maybe the point is moot, since `load` can be refactored to eliminate `withTempName`.

Reply:
I suppose we can use the same tmpFile approach for load.